### PR TITLE
Hardware requirements checks

### PR DESCRIPTION
**Pull request by ssd04**
## Reasoning behind the pull request
- Check cpu flags checks

## Testing procedure
- Check if node will fail to start if no required cpu flags available
- Standard system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by iulianpascalau — DISMISSED** [review]

**Review by sstanculeanu — DISMISSED** [review]

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: v1.6.11-dev-config-ace90ef010 -> hardware-requirements-chec-639da41b11

--- Specific errors ---

block hash does not match 6352
wrong nonce in block 2372
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  0
Nr. of all WARNS:  951
Nr. of new ERRORS:  0
Nr. of new WARNS:  1
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

--- WARNINGS ---

upcloud-fra-observer-2 :
Warn: wt.Listen()-> connection problem     error = connection not open  1

/------/

File: cmd/node/main.go [hunk 1]
```diff
@@ -301,3 +307,11 @@ func attachFileLogger(log logger.Logger, flagsConfig *config.ContextFlagsConfig)
 
 	return fileLogging, nil
 }
+
+func checkHardwareRequirements() error {
+	if !cpuid.CPU.Supports(cpuid.SSE4, cpuid.SSE42) {
```
Comment (iulianpascalau) [inline cmd/node/main.go:312]: I do not like having these hardcoded, can we somehow extract them in config.toml ?

File: cmd/node/main.go [hunk 2]
```diff
@@ -308,10 +308,32 @@ func attachFileLogger(log logger.Logger, flagsConfig *config.ContextFlagsConfig)
...
 		return fmt.Errorf("CPU Flags: Streaming SIMD Extensions 4 required")
 	}
 
 	return nil
 }
+
+func parseFeatures(features []string) ([]cpuid.FeatureID, error) {
+	flags := make([]cpuid.FeatureID, 0)
+
+	for _, cpuFlag := range features {
```
Comment (iulianpascalau) [inline cmd/node/main.go:327]: no need for this for range, we have the `ParseFeature` function

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: cmd/node/main.go (hunk 2 above)
Comment (ssd04) [reply cmd/node/main.go:327]: updated to use ParseFeature func

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: v1.6.11-dev-config-ace90ef010 -> hardware-requirements-chec-728c22bfa4

--- Specific errors ---

block hash does not match 10590
wrong nonce in block 3751
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  17
Nr. of all WARNS:  4310
Nr. of new ERRORS:  17
Nr. of new WARNS:  9
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

**Pull request merged**